### PR TITLE
BAH-2594 | Adding migrations to create role for accessing teleconsult…

### PR DESCRIPTION
**Pull request by kushboojain**
…ation module

### CODE REVIEW

File: bahmnicore-omod/src/main/resources/liquibase.xml [hunk 1]
```diff
@@ -4727,5 +4727,35 @@
...
+                SELECT count(*) FROM role WHERE role = 'Teleconsultation:CreateLink';
+            </sqlCheck>
+        </preConditions>
+        <comment>Creating new role Teleconsultation:CreateLink for Teleconsultation module</comment>
+        <sql>
+            INSERT INTO role(role, description, uuid) VALUES ("Teleconsultation:CreateLink", "Ability to create teleconsultation link in teleconsultation module", uuid());
+        </sql>
+    </changeSet>
+
+    <changeSet id="adding-create-teleconsultation-privilege-to-teleconsultation-createLink-role-202212012000" author="Kushboo">
```
Comment (gsluthra) [inline bahmnicore-omod/src/main/resources/liquibase.xml:4742]: author = bahmni

File: bahmnicore-omod/src/main/resources/liquibase.xml [hunk 2]
```diff
@@ -4727,5 +4727,35 @@
...
+        <preConditions onFail="CONTINUE">
+            <sqlCheck expectedResult="1">
+                SELECT count(*) FROM privilege WHERE privilege = "Create Teleconsultation";
+            </sqlCheck>
+            <sqlCheck expectedResult="1">
+                SELECT count(*) FROM role WHERE role = "Appointments:ReadOnly";
```
Comment (gsluthra) [inline bahmnicore-omod/src/main/resources/liquibase.xml:4748]: Why is this check here? Appointment:ReadOnly?

File: bahmnicore-omod/src/main/resources/liquibase.xml [hunk 3]
```diff
@@ -4727,5 +4727,35 @@
...
 
+    <changeSet id="create-teleconsultation-link-role-202212012000" author="Kushboo">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+                SELECT count(*) FROM role WHERE role = 'Teleconsultation:CreateLink';
+            </sqlCheck>
+        </preConditions>
+        <comment>Creating new role Teleconsultation:CreateLink for Teleconsultation module</comment>
+        <sql>
+            INSERT INTO role(role, description, uuid) VALUES ("Teleconsultation:CreateLink", "Ability to create teleconsultation link in teleconsultation module", uuid());
```
Comment (gsluthra) [inline bahmnicore-omod/src/main/resources/liquibase.xml:4738]: If someone has the ability to create an Appointment, then I think they can create a TeleCon appointment too. Should not need a new role. This privilege can be added in createAppointment Role itself.

I am not sure what is the need for a new role?

**Review by gsluthra — CHANGES_REQUESTED** [review]

File: bahmnicore-omod/src/main/resources/liquibase.xml (hunk 2 above)
Comment (kushboojain) [reply bahmnicore-omod/src/main/resources/liquibase.xml]: Sorry, it was a mistake. removed

File: bahmnicore-omod/src/main/resources/liquibase.xml (hunk 3 above)
Comment (kushboojain) [reply bahmnicore-omod/src/main/resources/liquibase.xml:4738]: Updated, now appointments role has create teleconsultation privilege

File: bahmnicore-omod/src/main/resources/liquibase.xml [hunk 4]
```diff
@@ -4727,5 +4727,29 @@
...
 
+    <changeSet id="create-teleconsultation-link-role-202212012000" author="BAHMNI">
+        <preConditions onFail="MARK_RAN">
+            <sqlCheck expectedResult="0">
+                SELECT count(*) FROM role WHERE role = 'Teleconsultation:CreateLink';
+            </sqlCheck>
+        </preConditions>
+        <comment>Creating new role Teleconsultation:CreateLink for Teleconsultation module</comment>
+        <sql>
+            INSERT INTO role(role, description, uuid) VALUES ("Teleconsultation:CreateLink", "Ability to create teleconsultation link in teleconsultation module", uuid());
```
Comment (gsluthra) [inline bahmnicore-omod/src/main/resources/liquibase.xml:4738]: Why is this role needed?

Comment (binduak) [reply bahmnicore-omod/src/main/resources/liquibase.xml:4738]: @kushboojain does this role is required here ?

Comment (kushboojain) [reply bahmnicore-omod/src/main/resources/liquibase.xml]: Removed

**Review by binduak — APPROVED** [review]

**Pull request merged**